src/modules/DecodeQr/Module.js

```diff
@@ -15,19 +15,17 @@ module.exports = function DoNothing(options, UI) {
 
     var step = this;
 
-    getPixels(input.src, function(err, pixels) {
-
-      if (err) throw err;
-
+    function extraManipulation(pixels){
       var w = pixels.shape[0];
       var h = pixels.shape[1];
       var decoded = jsQR(pixels.data, w, h);
-
-
-      // Tell Image Sequencer that this step is complete
-      options.step.qrval = (decoded) ? decoded.data : 'undefined';
-    });
-
+      var qrValue = (decoded) ? decoded.data : 'undefined';
+      if (options.step.inBrowser && options.step.ui)
+        $(options.step.ui).find('.details').append('<p><b>Qr Code</b> : ' + qrValue + '</p>');
```

Comment:
Please don't manipulate the UI from the source. Instead, just update `options.step.qrval`. It will automatically update.

Reply:
I set it back to use qrValue though, but the Average Module, is also edditing the UI from the source, should I create a seperate PR to fix that?

Reply:
Open an issue as of now. If you want to fix it, then fix it too.

Reply:
on it